### PR TITLE
Add authentication for diamondash api client requests

### PR DESCRIPTION
**Pull request by justinvdm**

### CODE REVIEW

File: go/dashboard/client.py [hunk 1]
```diff
@@ -17,8 +17,17 @@ def make_api_url(self, path):
             p.strip('/')
             for p in [settings.DIAMONDASH_API_URL, path])
 
+    def get_api_auth(self):
+        return (
+            settings.DIAMONDASH_API_USERNAME,
+            settings.DIAMONDASH_API_PASSWORD)
```
Comment (hodgestar) [inline go/dashboard/client.py]: Maybe we should return `None` if this is not set? That way AUTH is optional.

**Comment by hodgestar** [comment]
Other than the suggestion to make auth optional, +1.

File: go/dashboard/client.py (hunk 1 above)
Comment (justinvdm) [inline go/dashboard/client.py]: Good idea

**Comment by justinvdm** [comment]
Ready for review.

**Comment by justinvdm** [comment]
Ready for another review.

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**